autosklearn/pipeline/components/data_preprocessing/feature_type.py

```diff
@@ -139,33 +155,49 @@ def fit(
             else:
                 columns = set(range(n_feats))
             if expected != columns:
-                raise ValueError(
-                    f"Train data has columns={expected} yet the"
-                    f" feat_types are feat={columns}"
-                )
+                try:
+                    columns = [str(col) for col in columns]
+                except Exception as e:
+                    raise ValueError(
+                        f"Train data has columns={expected} yet the"
+                        f" feat_types are feat={columns}\n"
+                        f"Exception: {e}"
+                    )
+            transformer_lst = []
+
             categorical_features = [
                 key
                 for key, value in self.feat_type.items()
                 if value.lower() == "categorical"
             ]
+            if len(categorical_features) > 0:
+                transformer_lst.append(
+                    ("categorical_transformer", self.categ_ppl, categorical_features)
+                )
+
             numerical_features = [
                 key
                 for key, value in self.feat_type.items()
                 if value.lower() == "numerical"
             ]
+            if len(numerical_features) > 0:
+                transformer_lst.append(
+                    ("numerical_transformer", self.numer_ppl, numerical_features)
+                )
+
             text_features = [
                 key
                 for key, value in self.feat_type.items()
                 if value.lower() == "string"
             ]
+            if len(text_features) > 0:
+                transformer_lst.append(
+                    ("text_transformer", self.txt_ppl, text_features)
+                )
 
             sklearn_transf_spec = [
                 (name, transformer, feature_columns)
-                for name, transformer, feature_columns in [
-                    ("categorical_transformer", self.categ_ppl, categorical_features),
-                    ("numerical_transformer", self.numer_ppl, numerical_features),
-                    ("text_transformer", self.txt_ppl, text_features),
-                ]
+                for name, transformer, feature_columns in transformer_lst
```

Comment:
Is this for-loop still necessary given the changes you did above? Or could you revert the changes above?